service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java

```diff
@@ -167,6 +170,39 @@ public ResponseEntity<ApiAzureRelayNamespaceResource> getAzureRelayNamespace(
     return new ResponseEntity<>(resource.toApiResource(), HttpStatus.OK);
   }
 
+  @Override
+  public ResponseEntity<ApiCreatedAzureStorageContainerSasToken>
+      createAzureStorageContainerSasToken(UUID workspaceUuid, UUID storageContainerUuid) {
+    features.azureEnabledCheck();
+
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    final String userEmail =
```

Comment:
Why do we get the userEmail? Is it because we really want to log the email, or because we are using the Sam call as a permissions check?

Reply:
Just to get the `userEmail`. Logging the requesting user was called out in the ticket and the email wasn't available in the `AuthenticatedUserRequest` during my testing. It is a bit of a shame to have to make an extra call to Sam just to log the email so I'd be happy to pull it out if logging the user email is a soft requirement

Reply:
Hmmm, it seems worth checking into this requirement. Would it be possible to log something else available in the user request that would allow us to retrieve the email from Sam if necessary (if required to debug a user issue or investigate suspicious activity)?

Reply:
Do we need additional logging that WSM's Apache proxy doesn't give us? It seems like that would log all the same information we're logging here except for expiry time on the successful requests

Reply:
@marctalbott I'm not sure -- what identifier does the proxy log that will let us trace back to the calling principal?

Reply:
@aherbst-broad I'm not sure what it looks like for Azure, but for Google users we log email and Google subject id

It would look something like this for the SAS token endpoint
```
... 106960538915967416130 "marc.testbott@gmail.com" "-" [2022-06-13T14:34:42.987-0400] 3423867 + "POST /api/workspaces/v1/{workspace_id}/resources/controlled/azure/storageContainer/{resource_id}/getSasToken HTTP/1.1" 200 ...
```

Reply:
I am fine with leaving the Sam call in for now. If it ends up causing pain we can take a look at what is available in the Apache proxy for Azure users and remove the Sam call if possible at that time.